### DISCUSSION TITLE
Show specific KPI over generations

### DISCUSSION THREAD

**Discussion by stevanzou** [anyoptimization/pymoo#560] [Q&A] [answered]
Hi,

I have a question about the callback function. With the example from the callback, we can plot the best KPI for every generation. My goal is to observe one object and see how its KPI develops over generations.
Can I assume that `algorithm.pop.get("F")[0]` always represents the same object?

My guess is that it's not the same, but is there a possible method or workaround?
Thanks in advance and thanks for this nice framework.

Implementation:
algo = NSGA2
param = 4
KPIs = 3
problem=ElementwiseProblem

```
class MyCallback(Callback):
    def __init__(self) -> None:
        super().__init__()
        self.y1 = []
        self.y2 = []
        self.y3 = []

    def notify(self, algorithm):
        self.y1.append(algorithm.pop.get("F")[0][0])
        self.y2.append(algorithm.pop.get("F")[0][1])
        self.y3.append(algorithm.pop.get("F")[0][2])
```

**Comment by blankjul** [comment]
Any specific reason why you only look at the first solution in the population?
When you refer to KPI, do you refer to the smallest value for each objective?

If yes this is simply `np.min(algorithm.pop.get("F"), axis=0)`. Otherwise, yes you can log this in a callback function.

**Comment by stevanzou** [comment]
Let's say I have a machine with its parameters (height, weight, ...) and, by testing this machine, I receive the KPIs (speed, power consumption, ...). With the change of the parameters, the performance of this machine changes over the generations. Now, my thought would be to plot the speed development of this machine.

**Reply by blankjul** [reply, marked as answer]
I am not sure if I am able to grasp the problem. In the callback you have access to the full population and you can store as well as plot any type of statistics during the optimization run.

**Reply by stevanzou** [reply]
Now my problem seems clearer. We confused us in the team :) . Thank you so much for your help.
